pkg/controller/operators/olm/operator.go

```diff
@@ -827,6 +828,42 @@ func (a *Operator) syncNamespace(obj interface{}) error {
 		return err
 	}
 
+	// Filter the operator associated with this namespace
+	groups := make([]*v1.OperatorGroup, 0)
+	for _, og := range operatorGroupList {
+		if og.GetNamespace() == namespace.GetName() {
+			groups = append(groups, og)
+		}
+	}
+
+	// Check if there is a stale multiple OG condition and clear it if existed.
+	if len(groups) == 1 {
+		og := groups[0]
+		if c := meta.FindStatusCondition(og.Status.Conditions, v1.MutlipleOperatorGroupCondition); c != nil {
+			meta.RemoveStatusCondition(&og.Status.Conditions, v1.MutlipleOperatorGroupCondition)
+			_, err = a.client.OperatorsV1().OperatorGroups(namespace.GetName()).UpdateStatus(context.TODO(), og, metav1.UpdateOptions{})
+			if err != nil {
+				logger.Warnf("fail to upgrade operator group status condition og=%s: %s", og.GetName(), err.Error())
+			}
+		}
+	} else if len(groups) > 1 {
+		// Add to all OG's status conditions to indicate they're multiple OGs in the
+		// same namespace which is not allowed.
+		cond := metav1.Condition{
+			Type:    v1.MutlipleOperatorGroupCondition,
+			Status:  metav1.ConditionTrue,
+			Reason:  v1.MultipleOperatorGroupsReason,
+			Message: "Multiple OperatorGroup found in the same namespace",
+		}
+		for _, og := range groups {
+			meta.SetStatusCondition(&og.Status.Conditions, cond)
+			_, err = a.client.OperatorsV1().OperatorGroups(namespace.GetName()).UpdateStatus(context.TODO(), og, metav1.UpdateOptions{})
+			if err != nil {
+				logger.Warnf("fail to upgrade operator group status condition og=%s: %s", og.GetName(), err.Error())
+			}
```

Comment:
```suggestion
if _, err = a.client.OperatorsV1().OperatorGroups(namespace.GetName()).UpdateStatus(context.TODO(), og, metav1.UpdateOptions{}); err != nil {
	logger.Warnf("error updating OperatorGroup status condition og=%s: %s", in.GetName(), err.Error())
}
```
It may be worthwhile to log the condition as well.